flow/src/main/java/flow/ActivityFlowSupport.java

```diff
@@ -0,0 +1,133 @@
+ *   }
+ *
+ *   &#064;Override public void onBackPressed() {
+ *     if (!activityFlowSupport.onBackPressed()) {
+ *       super.onBackPressed();
+ *     }
+ *   }
+ *
+ *   &#064;Override public Object getSystemService(String name) {
+ *     Object service = activityFlowSupport.getSystemService(name);
+ *     return service != null ? service : super.getSystemService(name);
+ *   }
+ * }
+ * }</pre>
+ */
+public final class ActivityFlowSupport {
+  private static final String FLOW_KEY = ActivityFlowSupport.class.getSimpleName() + "_backstack";
+
+  private Parceler parceler;
+  private Flow.Dispatcher dispatcher;
+  private Flow flow;
+
+  public void onCreate(Bundle savedInstanceState, Parceler parceler, Backstack defaultBackstack) {
+    checkState(flow == null, "Flow already created.");
+    checkArgument(parceler != null, "parceler may not be null");
+    checkArgument(defaultBackstack != null, "defaultBackstack may not be null");
+    Backstack initialBackstack = null;
+    if (savedInstanceState != null && savedInstanceState.containsKey(FLOW_KEY)) {
+      initialBackstack = Backstack.from(savedInstanceState.getParcelable(FLOW_KEY), parceler);
+    }
+    if (initialBackstack == null) {
+      initialBackstack = defaultBackstack;
+    }
+    this.parceler = parceler;
+    flow = new Flow(initialBackstack);
+  }
+
+  public void onResume(Flow.Dispatcher dispatcher) {
+    checkState(flow != null, "Don't have a Flow. Did you forget to call onCreate()?");
+    checkArgument(dispatcher != null, "dispatcher may not be null");
+    this.dispatcher = dispatcher;
+    flow.setDispatcher(dispatcher);
+  }
+
+  public void onPause() {
+    checkState(flow != null, "Don't have a Flow. Did you forget to call onCreate()?");
+    checkState(dispatcher != null, "Don't have a Dispatcher. Did you forget to call onResume()?");
+    flow.removeDispatcher(dispatcher);
+    dispatcher = null;
+  }
+
+  /**
+   * @return true if the button press has been handled.
+   */
+  public boolean onBackPressed() {
+    checkState(flow != null, "Don't have a Flow. Did you forget to call onCreate()?");
+    return flow.goBack();
+  }
+
+  public void onSaveInstanceState(Bundle outState) {
+    checkArgument(outState != null, "outState may not be null");
+    checkState(flow != null, "Don't have a Flow. Did you forget to call onCreate()?");
+    Backstack backstack = getBackstackToSave(flow.getBackstack());
+    if (backstack == null) return;
+    //noinspection ConstantConditions
+    outState.putParcelable(FLOW_KEY, backstack.getParcelable(parceler));
+  }
+
+  /**
+   * @return The requested service, or null if it does not exist
+   */
+  public Object getSystemService(String name) {
+    if (Flow.isFlowSystemService(name)) {
+      checkState(flow != null, "Don't have a Flow. Did you forget to call onCreate()?");
+      return flow;
+    }
+    return null;
+  }
+
+  private Backstack getBackstackToSave(Backstack backstack) {
```

Comment:
Yeah, I'd like to punt this for now.  To lay the groundwork, I'm changing onCreate() into a factory for ActivitySupport.  If we want to add optional stuff like a persistence filter, I'm thinking we can add an `onCreate()` overload that returns a builder.